### PR TITLE
fix: multiselect relationship bug and improve accessibility

### PR DESCRIPTION
**Pull request by JessRynkar**
## Description

Closes [#117](https://github.com/payloadcms/payload-3.0-demo/issues/177) - hitting the space key while the `ReactSelect` is in focus crashes the page. 

This PR makes the following changes:
- Multivalue select component updated to only use `id`, drag feature does not work when using `uuid()`
- Ensures relationship field (multi and single value) can be accessed via the keyboard

- [X] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [X] Existing test suite passes locally with my changes

### CODE REVIEW

File: packages/ui/src/elements/ReactSelect/DropdownIndicator/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        if (e.key === 'Enter') {
+          // trigger the menu to open
+          return (e.key = ' ')
+        }
+      }}
+      role="button"
+      tabIndex={0}
+    >
+      <Chevron className={`${baseClass}__icon`} />
+    </div>
```
Comment (JarrodMFlesch) [inline packages/ui/src/elements/ReactSelect/DropdownIndicator/index.tsx:32]: I would rather the dropdown target element be a button than a div. See [here](https://github.com/payloadcms/payload/pull/6286/files#diff-ecc4ad56743dd2b308dbd9ce09258fde187ddc052f2d5d56c1fef55225eca855R16) on how to adjust the props type.

File: packages/ui/src/elements/ReactSelect/DropdownIndicator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div
+      className={baseClass}
+      ref={ref}
+      {...restInnerProps}
+      onKeyDown={(e) => {
+        if (e.key === 'Enter') {
+          // trigger the menu to open
+          return (e.key = ' ')
+        }
```
Comment (JarrodMFlesch) [inline packages/ui/src/elements/ReactSelect/DropdownIndicator/index.tsx:26]: I think you could just reassign the event to space in this case.

```ts
// allow for enter to also allow opening of the modal
if (e.key === 'Enter') e.key = ' '
```

in this case you should run theirs before yours.

**Review by JarrodMFlesch — CHANGES_REQUESTED** [review]
Couple other small things:
- Lets make the chevron stroke 1px when inside a relationship select to match the other icons
- I am unable to focus on the EditIcon and open the modal for editing when the relationship is a hasMany

**Comment by JessChowdhury** [comment]
@JarrodMFlesch pushed the following changes:
- change the chevron stroke width to 1px
- show focus outline for EditIcon when the relationship is a hasMany
- update the dropdown indicator to a button element
- simplify the drowdown indicator onClick as suggested

**Review by JarrodMFlesch — APPROVED** [review]

**Pull request merged**